x-pack/test/api_integration/apis/ml/jobs/datafeed_preview.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  const jobId = `fq_datafeed_preview_${Date.now()}`;
+
+  const job = {
+    job_id: `${jobId}_1`,
+    description: '',
+    groups: ['automated', 'farequote'],
+    analysis_config: {
+      bucket_span: '30m',
+      detectors: [{ function: 'distinct_count', field_name: 'airline' }],
+      influencers: [],
+    },
+    data_description: { time_field: '@timestamp' },
+    analysis_limits: { model_memory_limit: '11MB' },
+  };
+
+  function isUpperCase(str: string) {
+    return /^[A-Z]+$/.test(str);
+  }
+
+  function isLowerCase(str: string) {
+    return !/[A-Z]+/.test(str);
+  }
+
+  describe('Datafeed preview', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('ml/farequote');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    it(`should return a normal datafeed preview`, async () => {
+      const datafeed = {
+        datafeed_id: '',
+        job_id: '',
+        indices: ['ft_farequote'],
+        query: {
+          bool: {
+            must: [
+              {
+                match_all: {},
+              },
+            ],
+          },
+        },
+        runtime_mappings: {},
+      };
+
+      const { body } = await supertest
+        .post('/api/ml/jobs/datafeed_preview')
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send({ job, datafeed })
+        .expect(200);
+
+      expect(body.hits.total.value).to.eql(3207);
```

Comment:
If this validation fails it gives an error like
```
fail: apis Machine Learning jobs Datafeed preview should return a normal datafeed preview
      Error: expected 3208 to sort of equal 3207
      + expected - actual

      -3208
      +3207
```
This doesn't give much context when investigating a test failure, so I'd suggest to add a validation message, e.g.
```suggestion
      expect(body.hits.total.value).to.eql(3207, 'Response body total hits should be 3207');
```
This would give an error message like this:
```
fail: apis Machine Learning jobs Datafeed preview should return a normal datafeed preview
      Error: Response body total hits should be 3207
      + expected - actual

      -3208
      +3207
```

Reply:
Updated in https://github.com/elastic/kibana/pull/94760/commits/13eb4eb637c463c669bea9460385f17f0f79fcf6